src/solver/utils/include/antares/solver/utils/basis_status.h

```diff
@@ -0,0 +1,41 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+
+// NOTE
```

Comment:
Actually, since now we hide the status as implementation detail in this class, we could go further and just use our own enum here.
This way, we won't need to include `linear_solver.h` here either, this would be even more secure.

(this would also allow to use a plain BasisStatus member in optimization struct if we prefer)

Reply:
I disagree, conversion is painful and adds complexity and errors